Sources/Starknet/Network/StarknetBatchRequest.swift

```diff
@@ -0,0 +1,30 @@
+public struct StarknetBatchRequest<U: Decodable> {
+    let rpcPayloads: [JsonRpcPayload]
+    let config: HttpNetworkProvider.Configuration
+    let networkProvider: HttpNetworkProvider
+
+    public func send() async throws -> [Result<U, StarknetProviderError>] {
+        let rpcResponses: [JsonRpcResponse<U>] = try await networkProvider.send(
+            payload: rpcPayloads,
+            config: config,
+            receive: [JsonRpcResponse<U>.self]
+        )
+
+        return getOrderedRpcResults(rpcResponses: rpcResponses, count: rpcPayloads.count)
+    }
+}
+
+func getOrderedRpcResults<U: Decodable>(rpcResponses: [JsonRpcResponse<U>], count: Int) -> [Result<U, StarknetProviderError>] {
```

Comment:
I renamed it to `orderRpcResults`. It wasn't inlined on purpose, because the function is used in [testBatchResponseWithIncorrectOrder](https://github.com/software-mansion/starknet.swift/blob/5cfd5366c9a8179f7e9e48ea06c69cea5e5c5698/Tests/StarknetTests/Data/JsonRpcResponseTests.swift#L146)